### ISSUE TITLE
Help with BIOMOD_xxx - [changing from 3.5.1 functions]

### ISSUE THREAD

**Issue by ReproducibleScienceAssessment** [biomodhub/biomod2#553]
Can anyone help with changing code for the BIOMOD_ModelingOptions function so it is compatible with the current biomod2 updates (needs a new function name)

This was my previous code used to build the model however BIOMOD_ModelingOptions is no longer in function. 

BiomodOption <- BIOMOD_ModelingOptions(ANN = list(NbCV = 5, size = NULL, decay = NULL, rang = 0.1, maxit = 200),   # Artificial Neural Network (nnet)
                                       CTA = list(method = 'class', parms = 'default', cost = NULL),   # Classification Tree Analysis (rpart)
                                       FDA = list(method = 'mars', add_args = NULL),   #  Flexible Discriminant Analysis (fda)
                                       GBM = list(distribution = 'bernoulli', n.trees = 2500, shrinkage = 0.001,   
                                                  train.fraction = 1, cv.folds = 3, perf.method = 'cv', n.cores = 1),
                                       GLM = list(type = "quadratic", interaction.level = 0, myFormula = NULL,
                                                family = binomial(link = "logit"), test = "AIC"),
                                       GAM = list(algo = "GAM_mgcv", type = "s_smoother", k = 4, interaction.level = 0, 
                                                myFormula = NULL, family = binomial(link = "logit")),
                                       SRE = list(quant = 0.025),    # Surface Range Envelop (bm_SRE)
                                       MARS = list(myFormula = NULL, nk = NULL, penalty = 2, thresh = 0.001,  Splines (earth)
                                                   nprune = NULL, pmethod = "backward"),
                                       RF = list(do.classif = TRUE, ntree = 500, mtry = 'default', nodesize = 5,  
                                                 maxnodes = NULL),
                                       MAXENT.Phillips = list(path_to_maxent.jar = path_to_maxent,    
                                                            visible = FALSE, maximumiterations = 500, memory_allocated = 512,
                                                            product = FALSE, threshold = FALSE, hinge = FALSE))  

Any help would be massively appreciated!

**Comment by HeleneBlt** [comment]
Hello there,

Indeed we have made some changes since the version you use: now you can specify different options for each run ! We also use the real names of the arguments for each algorithms 💪 
You can find more information about it in the documentation of [bm_ModelingOptions](https://biomodhub.github.io/biomod2/reference/bm_ModelingOptions.html) and in the [Options vignette](https://biomodhub.github.io/biomod2/articles/vignette_modelingOptions.html). 

You code will look like this :

```
## In the case of real absences with two RUN

AnnOptions <- list('_allData_RUN1' = list(size = NULL, decay = NULL, rang = 0.1, maxit = 200),
                   '_allData_RUN2' = list(size = NULL, decay = NULL, rang = 0.1, maxit = 200))

# Or with rep
myRunNames <- paste0("_allData_RUN", 1:3, sep = "")

GAMOptions <- list(formula = bm_MakeFormula(resp.name = myBiomodData@sp.name, expl.var = head(myBiomodData@data.env.var), type = "s_smoother", k = 4, interaction.level = 0), family = binomial(link = "logit")),
GAMOptions <- rep(list(GAMOptions), length(myRunNames))
names(GAMOptions) <- myRunNames

# In the version 4.2-6-2, we add a special feature to change all the datasets in the same time

RFOptions <- list("for_all_datasets" = list(ntree = 500, mtry = 'default', nodesize = 5, maxnodes = NULL))
MAXENTOptions <- list("for_all_datasets" = list(path_to_maxent.jar = path_to_maxent,
                                                visible = FALSE, maximumiterations = 500, memory_allocated = 512,
                                                product = FALSE, threshold = FALSE, hinge = FALSE))

## After you can create your user.val 

user.val <- list(ANN.binary.nnet.nnet = AnnOptions,
                 GAM.binary.mgcv.gam = GAMOptions,
                 MAXENT.binary.MAXENT.MAXENT = MAXENTOptions,
                 RF.binary.randomForest.randomForest = RFOptions)

myOpt <- bm_ModelingOptions(data.type = 'binary',
                            models = yourmodels,
                            strategy = "user.defined",
                            user.val = user.val,
                            user.base = "bigboss",
                            bm.format = myBiomodData, 
                            calib.lines = myCVtable)
```

Don't hesitate if it's not clear!
Hélène

**Comment by HeleneBlt** [comment]
Hi Sky ! 

There is a little misunderstanding here: 

You need to indicate the name of the datasets you want for each options set. 
A dataset name will be of the form of **_PAi_RUNj** with i and j, the integer indicating the pseudo-absences dataset or the cross validation dataset.
If you doesn't have pseudo_absences but real absences, it will be **_allData_RUNj**. 
If you don't provide any cross validation table (with `calib.lines` argument), it will be **_allData_allRun** or **_PAi_allRun**. 

If you use `biomod2 version 4.2-6-2` ( ❗ and not the older versions ❗ : I'm not sure that the version you use), you could use `"for_all_datasets"` which will attribute the options set to all the datasets. 

So here it seems you only have one pseudo absences dataset, so you code will be : 

```
myRunNames <- c('_allData_allRun', '_PA1_allRun')
GAMOptions <- list(formula = bm_MakeFormula(resp.name = BiomodData@sp.name, expl.var = head(BiomodData@data.env.var), type = "s_smoother", k = 4, interaction.level = 0), family = binomial(link = "logit"))
GAMOptions <- rep(list(GAMOptions), length(myRunNames))
names(GAMOptions) <- myRunNames

GLMOptions <- list(formula = bm_MakeFormula(resp.name = BiomodData@sp.name, expl.var = head(BiomodData@data.env.var, type = "quadratic", interaction.level = 0, family = binomial(link = "logit"))))
GLMOptions <- rep(list(GLMOptions), length(myRunNames))
names(GLMOptions) <- myRunNames


RFOptions <- list(ntree = 500, mtry = 'default', nodesize = 5, maxnodes = NULL)
RFOptions <- rep(list(RFOptions), length(myRunNames))
names(RFOptions) <- myRunNames


MAXENTOptions <- list(path_to_maxent.jar = path_to_maxent, visible = FALSE, maximumiterations = 500,
                      memory_allocated = 512, product = FALSE, threshold = FALSE, hinge = FALSE)
MAXENTOptions <- rep(list(MAXENTOptions), length(myRunNames))
names(MAXENTOptions) <- myRunNames
```


Note that you can check our last [tutorial video](https://biomodhub.github.io/biomod2/articles/vignette_videos.html) at time 19:49 introducing this. 🖼️ 

Hope it's clearer that way ! 🙂 
Hélène

**Comment by ReproducibleScienceAssessment** [comment]
Hiya, 

Thats amazing! Its worked thank you. 

Unfortunately I'm now running my models and getting this error:

Creating Maxent Temp Proj Data...
*** Error in MAXENT, more info available in  ./EidolonDupreanum/models/EidolonDupreanum10models/EidolonDupreanum_allData_allRun_MAXENT_outputs/maxent.stderr
*** inherits(g.pred,'try-error')
   ! Note :  EidolonDupreanum_allData_allRun_MAXENT failed!

The more info available is: Error: Unable to access jarfile /Users/skystewart-roberts/Documents/OneDrive

My code to generate the models is:

BiomodModelOut <- BIOMOD_Modeling(bm.format = BiomodData, models = c('GLM', 'GAM', 'RF', 'MAXENT'),
                                    CV.nb.rep = 4,   # specify four-fold cross-validation
                                    CV.perc = 0.8, prevalence = 0.5, var.import = 3,
                                    metric.eval = c('KAPPA', 'TSS', 'ROC'),
                                    scale.models = FALSE, CV.do.full.models = TRUE,   # best settings for rare species
                                    modeling.id = paste(myRespName,"10models",sep=""))

With my MAXENT input coming from:

path_to_maxent = file.path( "~/Documents/OneDrive - Nexus365/Biology/Year 4/Coding/Malagasy_Bat_Data/maxent.jar")
path_to_maxent= paste(system.file(package="dismo"), "/java", sep='')

file.exists(path_to_maxent)
system("java -version")

I have checked the file exists and the file directory goes straight to a maxent.jar file. I've seen others have had previous problems but none of their solutions seem to be working. 

Any extra help would be much appreciated! 

Thank you, 

Sky

**Comment by HeleneBlt** [comment]
Hi Sky ! 

My first guess will be that Java has some troubles with " - " in the path file. Could you try to change the path or move the `maxent.jar` file somewhere with a simpler path ? 

Thanks 
Hélène

**Comment by ReproducibleScienceAssessment** [comment]
I have changed the name so it is no longer a '-' but still getting the same error.

**Comment by HeleneBlt** [comment]
Do you have access to the interface of Maxent when you do: 

`system("java -jar  OneDrive - Nexus365/Biology/Year 4/Coding/Malagasy_Bat_Data/maxent.jar") `


And secondly, when you give the path to `path_to_maxent.jar`, did you give : 

1. "~/Documents/OneDrive - Nexus365/Biology/Year 4/Coding/Malagasy_Bat_Data/maxent.jar"
or 
2. "~/Documents/OneDrive - Nexus365/Biology/Year 4/Coding/Malagasy_Bat_Data"

The good answer is 2 here ! 

Hélène

**Comment by ReproducibleScienceAssessment** [comment]
Here is my file path for MAXENT:
path_to_maxent.jar = file.path( "~/Documents/OneDrive_Nexus365/Biology/Year4/Coding/Malagasy_Bat_Data/maxent.jar")
path_to_maxent= paste(system.file(package="dismo"), "/java", sep='')

When i try and access the interface it says it cannot access the jarfile? Using this code : system('java -jar OneDrive_Nexus365/Biology/Year4/Coding/Malagasy_Bat_Data/maxent.jar')
